src/torchcodec/samplers/_implem.py

```diff
@@ -89,32 +90,37 @@ def _get_clip_span(*, num_indices_between_frames, num_frames_per_clip):
     return num_indices_between_frames * (num_frames_per_clip - 1) + 1
 
 
+_LIST_OF_INT_OR_FLOAT = Union[list[int], list[float]]
```

Comment:
Random aside: since we bumped our minimum Python version up to 3.9, I think we can use the `|` operator to indicate unions now. This would apply to a lot of other places as well. There's also some other type hint nicetities I think we can take advatnage of, like (I think) not needing `typing.List` and `typing.Dict` anymore. Not worth dealing with in this PR, I'm just making the comment to raise awareness for some code cleanup efforts in the future.

Reply:
Ah, I tried but sadly the use of `|` is 3.10 https://peps.python.org/pep-0604/

Seems like we can still use the modern `|`  if we do `from __future__ import annotations` https://stackoverflow.com/questions/76579085/python-3-9-use-or-operator-for-union-types